File: NuGen/Services/RandomGenerator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using NuGen.Services.Interfaces;

namespace NuGen.Services
{
    public class RandomGenerator : IRandomGenerator
    {
        private readonly Random _random = new();
        private readonly IUniqCheckService _check;

        public RandomGenerator(IUniqCheckService check)
        {
            _check = check;
        }

        public async IAsyncEnumerable<long> GenerateUniqueNumbersAsync(long from, long to)
        {
            for (long i = from; i <= to; i++)
            {
                while (true)
                {
                    long valueToAdd = _random.Next(0, 999999);
                    if (!await _check.CheckUniquenessAsync(valueToAdd))
                    {
                        yield return valueToAdd;
                    }
                    else
                    {
                        break;
                    }
                }
            }
        }
    }
}

File: NuGen.Services.Fsharp/RandomGenerator.fs
namespace NuGen.Services.Fsharp

open System
open Microsoft.Extensions.Options
open NuGen.Options.Start
open NuGen.Services.Interfaces

type RandomGenerator(check: IUniqCheckService, opts: IOptions<StartOptions>) =
    let rnd = Random()

    interface IRandomGeneratorService with
        member this.GenerateUniqueNumbersAsync(beginNumber, endNumber) =
            let maxNumber =
                int (
                    Math.Pow(10.0, float opts.Value.NumberOfDigits)
                    - 1.0
                )

            for i in opts.Value.From.Value .. opts.Value.To.Value do
                let value = rnd.Next(0, maxNumber)
                

            failwith "todo"
